playbooks/roles/airship-deploy-osh/tasks/main.yml

```diff
@@ -68,10 +75,16 @@
   tags:
     - install
 
+# TODO(aagate): Add a changed_when: to help idempotency
 - name: Collect site config
-  shell: 'IMAGE={{ pegleg_image }} TERM_OPTS="-l info" {{ pegleg }} site -r . collect {{ suse_airship_deploy_site_name }} -s {{ pegleg_output }}'
+  command: '{{ pegleg }} site -r . collect {{ suse_airship_deploy_site_name }} -s {{ pegleg_output }}'
   args:
     chdir: '{{ treasuremap_dir }}'
+  environment:
+    IMAGE: '{{ pegleg_image }}'
+    TERM_OPTS: '-l info'
+  tags:
```

Comment:
I suppose it's not possible to parse the output to have a `changed_when` and `failed_when` ?

Reply:
More correctly I am not sure all the permutations and combinations for a success or failure, hence added TODO